lexrpc/client.py

```diff
@@ -115,12 +115,12 @@ def __getattr__(self, attr):
 
         return getattr(super(), attr)
 
-    def call(self, nsid, input=None, **params):
+    def call(self, nsid, input=None, headers={}, **params):
```

Comment:
Instead of making callers pass `Content-Type` manually, let's figure it out automatically based on the method lexicon's encoding? ie something like `self._get_def(nsid)['input']['encoding']` below. If you get ambitious, you can also check that that matches `input`'s type.

Reply:
That only works if it's defined in the lexicon right? We won't have that info for blob upload, it's user-defined.

Reply:
Hmm, it is in the lexicon, right?

https://github.com/snarfed/lexrpc/blob/c5741011fcc3a07b35ef13d59ba72519e0dbfabf/lexrpc/lexicons/com/atproto/repo/uploadBlob.json#L1-L10

We're ok with not knowing the exact type, we can just see if it's `application/json` or not, right?

Reply:
I may be getting confused with what you mean but I don't think so, we actually need to set the correct header when uploading a blob, e.g. `image/png` or `image/jpeg`

Reply:
Ahh, sorry, I see what you mean now. True!